sdk/resource/resource.go

```diff
@@ -211,6 +196,18 @@ func Empty() *Resource {
 // Default returns an instance of Resource with a default
 // "service.name" and OpenTelemetrySDK attributes.
 func Default() *Resource {
+	if defaultResource == nil {
```

Comment:
It looks like it is possible for the return value of `Detect` to be nil.  At the end of this block please set `defaultResource = emptyResource` if that happens.  It should never happen with the set of detectors provided here but things change and that could cause this detection to run more than once per process.

Reply:
May want to also consider (overkill perhaps?) using a `sync.Once` to avoid races...